### PR TITLE
Forward ICMP packets to OVN loadbalancer

### PR DESCRIPTION
**Pull request by pperiyasamy**
This adds flow rule into external ovs bridge so that Destination unreachable (Fragmentation needed) packet can be put onto OVN pipeline for a given loadbalancer or ingress type of service ip address and forwards those packets into appropriate target endpoint pods. This eventually makes Path MTU Discovery (PMTUD) works end to end.

This is just an extension of [freedge](https://github.com/freedge)'s effort (PR https://github.com/ovn-org/ovn-kubernetes/pull/3696) and makes that work for every external service scenarios.

### CODE REVIEW

File: go-controller/pkg/node/gateway_shared_intf.go [hunk 1]
```diff
@@ -372,6 +375,15 @@ func (npw *nodePortWatcher) generateArpBypassFlow(protocol string, ipAddr string
 	return arpFlow
 }
 
+func (npw *nodePortWatcher) generateIcmpFragmentationFlow(nwDst, ipAddr string, cookie string) string {
+	// we send any ICMP destination unreachable, fragmentation needed to the OVN pipeline too so that
+	// path MTU discovery continues to work.
+	icmpFragmentationFlow := fmt.Sprintf("cookie=%s, priority=110, in_port=%s, icmp, %s=%s, icmp_type=3, icmp_code=4, "+
```
Comment (dcbw) [inline go-controller/pkg/node/gateway_shared_intf.go:381]: Do we also need ICMPv6 flows?

File: go-controller/pkg/node/gateway_shared_intf.go [hunk 2]
```diff
@@ -281,6 +281,9 @@ func (npw *nodePortWatcher) createLbAndExternalSvcFlows(service *kapi.Service, s
 	// add the ARP bypass flow regardless of service type or gateway modes since its applicable in all scenarios.
 	arpFlow := npw.generateArpBypassFlow(protocol, externalIPOrLBIngressIP, cookie)
 	externalIPFlows := []string{arpFlow}
+	// add the ICMP Fragmentation flow regardless of service type or gateway modes since its applicable in all scenarios.
+	icmpFlow := npw.generateIcmpFragmentationFlow(nwDst, externalIPOrLBIngressIP, cookie)
```
Comment (trozet) [inline go-controller/pkg/node/gateway_shared_intf.go:285]: I don't really understand this. A flow is being created to handle a packet destined to a service with needs frag, but how would that ever happen? It would require someone to send a packet out of the host with the external IP or LB ip as the source IP...which is not possible unless someone was using the egress service feature.

Also, why are we targeting services here. What about a regular pod that sends an egress packet for path MTU discovery? We have flows in table 1 to handle related traffic:

```
 cookie=0xdeff105, duration=9741.566s, table=1, n_packets=0, n_bytes=0, priority=100,ct_state=+est+trk,ct_mark=0x1,ip actions=output:"patch-breth0_ov"
 cookie=0xdeff105, duration=9741.566s, table=1, n_packets=2265247, n_bytes=216632842, priority=100,ct_state=+est+trk,ct_mark=0x2,ip actions=LOCAL
 cookie=0xdeff105, duration=9741.566s, table=1, n_packets=0, n_bytes=0, priority=100,ct_state=+rel+trk,ct_mark=0x1,ip actions=output:"patch-breth0_ov"
 cookie=0xdeff105, duration=9741.566s, table=1, n_packets=0, n_bytes=0, priority=100,ct_state=+rel+trk,ct_mark=0x2,ip actions=LOCAL
```

I'm not sure if conntrack considers ICMP needs frag as related or not, but I presume it probably does. If that is the case, any needs frag packet caused by something OVN sending PMTUD would be sent back to OVN.

What am I missing here?

Comment (trozet) [reply go-controller/pkg/node/gateway_shared_intf.go:285]: oh I guess there could be a UDP initiated connection to the load balancer IP from outside, then the reply is sent from a pod with DF, and that would cause the outside to send back needs frag, but still...wouldn't the related flow take care of this?

Comment (pperiyasamy) [reply go-controller/pkg/node/gateway_shared_intf.go:285]: yes @trozet, The ICMP frag packet is triggered by external client network towards service, but somehow the packet doesn't reach ovn pipeline thought it reaches br-ex. I'm not able to repro this case in shared gw mode on my kind cluster so far, would attempt again and see if those conntrack ip flows would take the packet into ovn pipeline.

File: go-controller/pkg/node/gateway_shared_intf.go (hunk 1 above)
Comment (pperiyasamy) [reply go-controller/pkg/node/gateway_shared_intf.go]: good catch, done.

File: go-controller/pkg/node/gateway_shared_intf.go (hunk 2 above)
Comment (tssurya) [reply go-controller/pkg/node/gateway_shared_intf.go:285]: conntrack considers ICMP as related traffic yes.

Comment (pperiyasamy) [reply go-controller/pkg/node/gateway_shared_intf.go:285]: OK @tssurya and @trozet , can you explain which table does update on the `ct_mark` field ? I don't see any of the table flows does that before sending the packet into table 1 which has match `ct_mark` . 

```
# ovs-ofctl -Oopenflow13 dump-flows breth0
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=0, n_bytes=0, priority=500,ip,in_port="patch-breth0_ov",nw_src=172.18.0.3,nw_dst=169.254.169.2 actions=ct(commit,table=4,zone=64001,nat(dst=172.18.0.3))
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=0, n_bytes=0, priority=500,ip,in_port=LOCAL,nw_dst=169.254.169.1 actions=ct(table=5,zone=64002,nat)
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=4422, n_bytes=387654, priority=500,ip,in_port=LOCAL,nw_dst=10.96.0.0/16 actions=ct(commit,table=2,zone=64001,nat(src=169.254.169.2))
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=0, n_bytes=0, priority=105,ip,in_port="patch-breth0_ov",nw_dst=10.96.0.0/16 actions=drop
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=4096, n_bytes=3255037, priority=500,ip,in_port="patch-breth0_ov",nw_src=10.96.0.0/16,nw_dst=169.254.169.2 actions=ct(table=3,zone=64001,nat)
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=62, n_bytes=8184, priority=205,udp,in_port=eth0,dl_dst=02:42:ac:12:00:03,tp_dst=6081 actions=LOCAL
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=0, n_bytes=0, priority=200,udp,in_port=eth0,tp_dst=6081 actions=NORMAL
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=27, n_bytes=3564, priority=200,udp,in_port=LOCAL,tp_dst=6081 actions=output:eth0
 cookie=0x7ecab03ea1702658, duration=6272.703s, table=0, n_packets=0, n_bytes=0, priority=110,tcp,in_port=eth0,tp_dst=30406 actions=output:"patch-breth0_ov"
 cookie=0x7ecab03ea1702658, duration=6272.703s, table=0, n_packets=0, n_bytes=0, priority=110,tcp,in_port="patch-breth0_ov",tp_src=30406 actions=output:eth0
 cookie=0xe08c0118ee86770e, duration=6272.703s, table=0, n_packets=0, n_bytes=0, priority=110,arp,in_port=eth0,arp_tpa=192.168.10.0,arp_op=1 actions=LOCAL
 cookie=0xe08c0118ee86770e, duration=6272.703s, table=0, n_packets=0, n_bytes=0, priority=110,icmp,in_port=eth0,nw_dst=192.168.10.0,icmp_type=3,icmp_code=4 actions=output:"patch-breth0_ov"
 cookie=0xe08c0118ee86770e, duration=6272.703s, table=0, n_packets=27, n_bytes=1998, priority=110,tcp,in_port=eth0,nw_dst=192.168.10.0,tp_dst=80 actions=output:"patch-breth0_ov"
 cookie=0xe08c0118ee86770e, duration=6272.703s, table=0, n_packets=62, n_bytes=4588, priority=110,tcp,in_port="patch-breth0_ov",nw_src=192.168.10.0,tp_src=80 actions=output:eth0
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=0, n_bytes=0, priority=109,ip,in_port="patch-breth0_ov",nw_src=10.244.0.0/24 actions=ct(commit,zone=64000,exec(set_field:0x1->ct_mark)),output:eth0
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=0, n_bytes=0, priority=105,pkt_mark=0x3f0,ip,in_port="patch-breth0_ov" actions=ct(commit,zone=64000,nat(src=172.18.0.3),exec(set_field:0x1->ct_mark)),output:eth0
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=0, n_bytes=0, priority=104,ip,in_port="patch-breth0_ov",nw_src=10.244.0.0/16 actions=drop
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=7421, n_bytes=645245, priority=100,ip,in_port="patch-breth0_ov" actions=ct(commit,zone=64000,exec(set_field:0x1->ct_mark)),output:eth0
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=56841, n_bytes=4914931, priority=100,ip,in_port=LOCAL actions=ct(commit,zone=64000,exec(set_field:0x2->ct_mark)),output:eth0
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=71226, n_bytes=86541666, priority=50,ip,in_port=eth0 actions=ct(table=1,zone=64000,nat)
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=12898, n_bytes=8661061, priority=10,in_port=eth0,dl_dst=02:42:ac:12:00:03 actions=output:"patch-breth0_ov",LOCAL
 cookie=0x0, duration=10023.122s, table=0, n_packets=15216, n_bytes=2316676, priority=0 actions=NORMAL
 cookie=0xdeff105, duration=10022.167s, table=1, n_packets=6661, n_bytes=3832449, priority=100,ct_state=+est+trk,ct_mark=0x1,ip actions=output:"patch-breth0_ov"
 cookie=0xdeff105, duration=10022.167s, table=1, n_packets=63586, n_bytes=82544412, priority=100,ct_state=+est+trk,ct_mark=0x2,ip actions=LOCAL
 cookie=0xdeff105, duration=10022.167s, table=1, n_packets=0, n_bytes=0, priority=100,ct_state=+rel+trk,ct_mark=0x1,ip actions=output:"patch-breth0_ov"
 cookie=0xdeff105, duration=10022.167s, table=1, n_packets=0, n_bytes=0, priority=100,ct_state=+rel+trk,ct_mark=0x2,ip actions=LOCAL
 cookie=0xdeff105, duration=10022.167s, table=1, n_packets=0, n_bytes=0, priority=13,udp,in_port=eth0,tp_dst=3784 actions=output:"patch-breth0_ov",LOCAL
 cookie=0xdeff105, duration=10022.167s, table=1, n_packets=957, n_bytes=145755, priority=10,dl_dst=02:42:ac:12:00:03 actions=LOCAL
 cookie=0xdeff105, duration=10022.167s, table=1, n_packets=0, n_bytes=0, priority=0 actions=NORMAL
 cookie=0xdeff105, duration=2.636s, table=2, n_packets=4422, n_bytes=387654, actions=set_field:02:42:ac:12:00:03->eth_dst,output:"patch-breth0_ov"
 cookie=0xdeff105, duration=2.636s, table=3, n_packets=4101, n_bytes=3255367, actions=move:NXM_OF_ETH_DST[]->NXM_OF_ETH_SRC[],set_field:02:42:ac:12:00:03->eth_dst,LOCAL
 cookie=0xdeff105, duration=10022.167s, table=4, n_packets=0, n_bytes=0, ip actions=ct(commit,table=3,zone=64002,nat(src=169.254.169.1))
 cookie=0xdeff105, duration=10022.167s, table=5, n_packets=0, n_bytes=0, ip actions=ct(commit,table=2,zone=64001,nat)
# 
```
I've tried to repro the issue with steps given [here](https://gist.github.com/karampok/805588c5f1f14bc6600935caf221ca5f), but [routes](https://gist.github.com/karampok/805588c5f1f14bc6600935caf221ca5f#make-sure-traffic-can-flow-back-to-client-172200016-is-the-clientnet-cidr) towards frr container for reverse traffic are ignored and ovs directly puts the traffic onto eth0 and gets lost in the docket network. so not able to induce PMTUD packet towards service (where things works fine in local gw mode). How do we solve this ? Please let me know.

Comment (freedge) [reply go-controller/pkg/node/gateway_shared_intf.go:285]: > We have flows in table 1 to handle related traffic

but they are not used though (at least in the case of gateway in shared mode, ETP: Local) as following two flows forward the traffic statelessly between eth0 and patch-breth0_ov
```
 cookie=0xe08c0118ee86770e, duration=6272.703s, table=0, n_packets=27, n_bytes=1998, priority=110,tcp,in_port=eth0,nw_dst=192.168.10.0,tp_dst=80 actions=output:"patch-breth0_ov"
 cookie=0xe08c0118ee86770e, duration=6272.703s, table=0, n_packets=62, n_bytes=4588, priority=110,tcp,in_port="patch-breth0_ov",nw_src=192.168.10.0,tp_src=80 actions=output:eth0
```

when an ICMP is sent back from a switch, only then this packet reaches table=1 through
```
 cookie=0xdeff105, duration=10022.167s, table=0, n_packets=71226, n_bytes=86541666, priority=50,ip,in_port=eth0 actions=ct(table=1,zone=64000,nat)
```
But that does not match any established conntrack (64000 is for egress traffic?) and so in table=1 this flow sends the traffic back to the network (I guess through)
```
 cookie=0xdeff105, duration=10022.167s, table=1, n_packets=957, n_bytes=145755, priority=10,dl_dst=02:42:ac:12:00:03 actions=LOCAL
```

Thank you!

Comment (trozet) [reply go-controller/pkg/node/gateway_shared_intf.go:285]: I see now. Yeah 64000 is for egress traffic. It is indeed bypassing conntrack cause of the service flows. I need to think a little bit more about if we want to generate more statless flows to handle this, or if we should conntrack this path as well for maybe handling other related traffic scenarios.

Comment (pperiyasamy) [reply go-controller/pkg/node/gateway_shared_intf.go:285]: @freedge I'm trying to repro the issue with your steps given [here](https://github.com/ovn-org/ovn-kubernetes/pull/3696#issue-1770444701), But i never able to induce PMTUD packet at step 7, curl command (`curl 192.168.10.0:80/big.iso -o big.iso`) always works from the host until it fails for MTU size 100 with `curl: (56) Recv failure: Connection reset by peer` error.

Here is the tcpdump at worker node hosting the pod (when MTU size 100):

```
# sudo nsenter -t 120297 -n tcpdump -n -i any -vvv icmp or host 192.168.10.0
dropped privs to tcpdump
tcpdump: listening on any, link-type LINUX_SLL (Linux cooked v1), capture size 262144 bytes
11:07:16.978767 IP (tos 0x0, ttl 63, id 31227, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.57.1.57692 > 192.168.10.0.http: Flags [S], cksum 0x24d5 (correct), seq 1208296625, win 64240, options [mss 1460,sackOK,TS val 4080804424 ecr 0,nop,wscale 7], length 0
11:07:16.980272 IP (tos 0x0, ttl 62, id 0, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.10.0.http > 192.168.57.1.57692: Flags [S.], cksum 0xd74c (correct), seq 3392078381, ack 1208296626, win 64704, options [mss 1360,sackOK,TS val 3453333976 ecr 4080804424,nop,wscale 7], length 0
11:07:16.980519 IP (tos 0x0, ttl 63, id 31228, offset 0, flags [DF], proto TCP (6), length 52)
    192.168.57.1.57692 > 192.168.10.0.http: Flags [.], cksum 0x007e (correct), seq 1, ack 1, win 502, options [nop,nop,TS val 4080804426 ecr 3453333976], length 0
11:07:17.187289 IP (tos 0x0, ttl 63, id 31230, offset 0, flags [+], proto TCP (6), length 100)
    192.168.57.1.57692 > 192.168.10.0.http: Flags [P.], seq 1:49, ack 1, win 502, options [nop,nop,TS val 4080804633 ecr 3453333976], length 48: HTTP, length: 48
	GET /big.iso HTTP/1.1
	Host: 192.168.10.0
	User-[!http]
11:07:17.187382 IP (tos 0x0, ttl 63, id 31230, offset 80, flags [none], proto TCP (6), length 55)
    192.168.57.1 > 192.168.10.0: ip-proto-6
11:07:17.189230 IP (tos 0x0, ttl 63, id 31230, offset 0, flags [+], proto TCP (6), length 100)
    192.168.57.1.57692 > 192.168.10.0.http: Flags [P.], seq 1:49, ack 1, win 502, options [nop,nop,TS val 4080804633 ecr 3453333976], length 48: HTTP, length: 48
	GET /big.iso HTTP/1.1
	Host: 192.168.10.0
	User-[!http]
11:07:17.189262 IP (tos 0x0, ttl 63, id 31230, offset 80, flags [none], proto TCP (6), length 55)
    192.168.57.1 > 192.168.10.0: ip-proto-6
11:07:17.193387 IP (tos 0x0, ttl 61, id 0, offset 0, flags [DF], proto TCP (6), length 40)
    192.168.10.0.http > 192.168.57.1.57692: Flags [R], cksum 0x3d84 (correct), seq 3392078382, win 0, length 0
11:07:17.193664 IP (tos 0x0, ttl 61, id 0, offset 0, flags [DF], proto TCP (6), length 40)
    192.168.10.0.http > 192.168.57.1.57692: Flags [R], cksum 0x3d84 (correct), seq 3392078382, win 0, length 0

```
Can you help me to repro the issue ?

Comment (freedge) [reply go-controller/pkg/node/gateway_shared_intf.go:285]: what kind of captures do you see with mtu=400? the http request should go through, and the reply should have packets lengths > 400 bytes, and in that case there should be some ICMP packets.
Here the http request does not go through, also it is fragmented, and the fragments do not reach the server. It's super weird. I would expect all packets to have the DF flag...

Comment (pperiyasamy) [reply go-controller/pkg/node/gateway_shared_intf.go:285]: @freedge I see reply has packet length almost upto maxmtu 65536 and seeing `net.ipv4.ip_no_pmtu_disc`  is also set with 0 on both Host and VM. Looks like mtu size does not come into effect. I'm using VirtualBox 7.0 running on Fedora 37 host.

Comment (pperiyasamy) [reply go-controller/pkg/node/gateway_shared_intf.go:285]: yes @freedge , now i can repro the issue within the VM using your example and verify the fix. Thank you. 
Now let me talk to @trozet for other related traffic scenarios.

Comment (pperiyasamy) [reply go-controller/pkg/node/gateway_shared_intf.go:285]: > I see now. Yeah 64000 is for egress traffic. It is indeed bypassing conntrack cause of the service flows. I need to think a little bit more about if we want to generate more statless flows to handle this, or if we should conntrack this path as well for maybe handling other related traffic scenarios.

@trozet The `ICMP Destination unreachable (Fragmentation needed)` packet would contain intermediate node/gateway IP address (not the actual client IP address) as src IP address, so can we still make it into conntrack path ?

Comment (trozet) [reply go-controller/pkg/node/gateway_shared_intf.go:285]: I would hope conntrack is able to still handle that, but should be easy to test it now that you have a reproducer. Try just modifying your service flow in table 0 to commit to conntrack zone 64000, load CT mark 1, then send it to table 1 and see what happens.

Comment (pperiyasamy) [reply go-controller/pkg/node/gateway_shared_intf.go:285]: Ok @trozet , I tried to change service flows like below:

```
cookie=0xe08c0118ee86770e, duration=3180.635s, table=0, n_packets=32, n_bytes=2368, priority=110,tcp,in_port=eth0,nw_dst=192.168.10.0,tp_dst=80 actions=ct(commit,table=1,zone=64000,exec(set_field:0x1->ct_mark))
cookie=0xe08c0118ee86770e, duration=3180.635s, table=0, n_packets=0, n_bytes=0, priority=110,tcp,in_port="patch-breth0_ov",nw_src=192.168.10.0,tp_src=80 actions=ct(commit,zone=64000,exec(set_field:0x1->ct_mark)),output:eth0
```
Am I correct ? With the above config, I don't see syn packet is getting into ovn pipeline, Looks like it's hitting on below table 1 flow entry and getting lost.
```
cookie=0xdeff105, duration=3510.318s, table=1, n_packets=371, n_bytes=63077, priority=10,dl_dst=02:42:ac:12:00:04 actions=LOCAL 
```
I still have the same question. I guess PMTUD packet is not associated with an established connection, how do we make it into conntrack path ?

Screenshot from 2023-08-23 13-32-12

Comment (pperiyasamy) [reply go-controller/pkg/node/gateway_shared_intf.go]: Yes @trozet , now added a new e2e test replicating @freedge's [procedure](https://github.com/ovn-org/ovn-kubernetes/pull/3762#discussion_r1297370672). Please review when you get time.

File: go-controller/pkg/node/gateway_shared_intf.go [hunk 3]
```diff
@@ -373,6 +376,23 @@ func (npw *nodePortWatcher) generateArpBypassFlow(protocol string, ipAddr string
 	return arpFlow
 }
 
+func (npw *nodePortWatcher) generateIcmpFragmentationFlow(nwDst, ipAddr string, cookie string) string {
```
Comment (trozet) [inline go-controller/pkg/node/gateway_shared_intf.go:379]: nit: keep the acronym all uppercase ICMPFragmentation

File: test/e2e/service.go [hunk 4]
```diff
@@ -873,6 +873,103 @@ spec:
...
+		err = buildAndRunCommand("sudo ip netns exec bridge ip link set bridgetoclient up")
+		framework.ExpectNoError(err, "failed to get bridgetoclient up")
+		err = buildAndRunCommand("sudo ip netns exec bridge ip addr add 192.168.223.1/24 dev bridgetoclient")
+		framework.ExpectNoError(err, "failed to add ip address on bridgetoclient gateway interface")
+		err = buildAndRunCommand("sudo ip netns exec client ip link set clienttobridge up")
+		framework.ExpectNoError(err, "failed to get clienttobridge up")
+		err = buildAndRunCommand("sudo ip netns exec client ip addr add 192.168.223.2/24 dev clienttobridge")
+		framework.ExpectNoError(err, "failed to add ip address on clienttobridge interface")
+		err = buildAndRunCommand("sudo ip netns exec client ip route add default via 192.168.223.1")
+		framework.ExpectNoError(err, "failed to add defaul route on client netns")
```
Comment (trozet) [inline test/e2e/service.go:939]: default

File: test/e2e/service.go [hunk 5]
```diff
@@ -873,6 +873,103 @@ spec:
...
+		err = buildAndRunCommand("sudo ip netns exec bridge ip route add default via 192.168.222.1")
+		framework.ExpectNoError(err, "failed to add defaul route on bridge netns")
```
Comment (trozet) [inline test/e2e/service.go:941]: ditto

File: test/e2e/service.go [hunk 6]
```diff
@@ -873,6 +873,103 @@ spec:
...
+		framework.ExpectNoError(err, fmt.Sprintf("failed to get endpoint's %s node ip address", endPointIP))
+
+		// The external client configuration done in install_metallb can not be used because routes for external client
+		// installed in K8s node https://github.com/ovn-org/ovn-kubernetes/blob/master/contrib/kind.sh#L1045-L1047
+		// are ignored in shared gateway mode and traffic coming back from pod is put on the docker bridge directly by
+		// br-ex flows which needs to be handled in host(or vm) network.
+		// Hence the following set of ip commands set up two networks called bridge (192.168.222.0/24) and
+		// client 192.168.223.0/24 on the host(or vm) machine. The external client (192.168.223.2) runs on the client
+		// network which tries to connect with load balancer service via bridge network. There would be also a route
+		// created for the load balancer service forwarding packet into the node which hosts one of the endpoint.
```
Comment (trozet) [inline test/e2e/service.go:899]: its hard to follow all of these ip commands and build the topology. Could you add an ascii diagram in here showing the topology with adjusted MTU and the connectivity the cluster?

File: test/scripts/e2e-cp.sh [hunk 7]
```diff
@@ -66,6 +66,13 @@ if [ "$OVN_GATEWAY_MODE" == "shared" ]; then
   SKIPPED_TESTS+="Should ensure load balancer service|LGW"
 fi
 
+if [ "$OVN_GATEWAY_MODE" == "local" ]; then
+  if [ "$SKIPPED_TESTS" != "" ]; then
+    SKIPPED_TESTS+="|"
+  fi
+  SKIPPED_TESTS+="Should ensure connectivity works on an external service when mtu changes in intermediate node"
```
Comment (trozet) [inline test/scripts/e2e-cp.sh:73]: why are we skipping it for local gw mode? Shouldn't it work there too?

**Review by trozet — CHANGES_REQUESTED** [review]

File: test/e2e/service.go (hunk 6 above)
Comment (pperiyasamy) [reply test/e2e/service.go:899]: oh! yes, added a diagram now. hope its ok.

File: test/scripts/e2e-cp.sh (hunk 7 above)
Comment (pperiyasamy) [reply test/scripts/e2e-cp.sh]: yes, right, it should work for local gw mode as well. 
removed from skip tests now.

**Review by trozet — APPROVED** [review]

File: test/e2e/service.go (hunk 6 above)
Comment (trozet) [reply test/e2e/service.go:899]: looks good, thank you
